simulation/traffic_simulator/src/entity/pedestrian_entity.cpp

```diff
@@ -129,7 +202,17 @@ void PedestrianEntity::setDecelerationLimit(double deceleration)
     THROW_SEMANTIC_ERROR("Deceleration limit should be over zero.");
   }
   auto behavior_parameter = getBehaviorParameter();
-  behavior_parameter.deceleration = deceleration;
+  behavior_parameter.dynamic_constraints.max_deceleration = deceleration;
+  setBehaviorParameter(behavior_parameter);
+}
+
+void PedestrianEntity::setDecelerationRateLimit(double deceleration_rate)
+{
+  if (deceleration_rate <= 0.0) {
+    THROW_SEMANTIC_ERROR("Deceleration rate limit should be over zero.");
+  }
+  auto behavior_parameter = getBehaviorParameter();
+  behavior_parameter.dynamic_constraints.max_deceleration_rate = deceleration_rate;
```

Comment:
Consider clamping the value to the range 0 to max_decleration_rate instead of raising an error.

Reply:
Deceleration value should not over 0, because it is physically impossible.
So I think raising error is best.